src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -133,7 +133,20 @@ private static IComponentActivator GetComponentActivatorOrDefault(IServiceProvid
     protected ComponentState GetComponentState(int componentId)
         => GetRequiredComponentState(componentId);
 
-    internal ComponentState GetComponentState(IComponent component)
+    /// <summary>
+    /// Gets the <see cref="IComponentRenderMode"/> for a given component if available.
+    /// </summary>
+    /// <param name="component">The component type</param>
+    /// <returns></returns>
+    protected internal virtual IComponentRenderMode? GetComponentRenderMode(IComponent component)
+        => null;
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="component"></param>
+    /// <returns></returns>
+    protected internal ComponentState GetComponentState(IComponent component)
```

Comment:
`EndpointHtmlRenderer` needs this API to find the component state from a callback target.